Automattic/jetpack
# Carousel: ensure a post is an attachment

Fixes an PHP warning: Warning: Invalid argument supplied for foreach() in /home/wpcom/public_html/wp-content/........carousel/jetpack-carousel.php on line 738 
#### Changes proposed in this Pull Request:

* Ensure that a post is an attachment before adding image data.

#### Other information:


#### Jetpack product discussion


n/a

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






* Setup Jetpack with Carousel enabled and set to automatically use the lightbox.
* Add a post with the content of `Thanks for joining me! <blockquote>Good company in a journey makes the way seem shorter. — Izaak Walton</blockquote><img 
class="wp-image-7 size-full" src="https://twentysixteendemo.files.wordpress.com/2015/11/post.png" alt="post" width="1000" height="563" />`
* Ensure there is a post ID 7 being used by a post/page/non-attachment.
* Load the page with that content (or run `the_content` filters on that text).
* Before patch, see PHP warning for lack of iterable. After, nothing in the log.
* Try a legit carousel and ensure it still works.
